### PR TITLE
Return ArrayBuffer, not view from concatenateArrayBuffers

### PR DESCRIPTION
**Pull request by kylebarron**
`concatenateArrayBuffers` currently returns a `Uint8Array` instead of an `ArrayBuffer`. This modifies that function to return an `ArrayBuffer`

### CODE REVIEW

File: modules/core/test/iterator-utils/make-iterator.spec.js [hunk 1]
```diff
@@ -26,9 +26,13 @@ test('concatenateChunksAsync', async t => {
...
   const arraybuffer = await concatenateChunksAsync(asyncArrayBuffers());
-  t.ok(arraybuffer instanceof Uint8Array, 'returns buffer');
+  t.ok(arraybuffer instanceof ArrayBuffer, 'returns ArrayBuffer');
   /* global TextEncoder */
-  t.deepEqual(arraybuffer, new TextEncoder().encode(RESULT), 'returns concatenated arraybuffer');
+  t.deepEqual(
+    arraybuffer,
+    new TextEncoder().encode(RESULT).buffer,
+    'returns concatenated ArrayBuffer'
+  );
```
Comment (kylebarron) [inline modules/core/test/iterator-utils/make-iterator.spec.js:35]: Should this test file be moved to `loader-utils` as well?

Comment (ibgreen) [reply modules/core/test/iterator-utils/make-iterator.spec.js:35]: Yes

Comment (ibgreen) [reply modules/core/test/iterator-utils/make-iterator.spec.js:35]: Yes

File: modules/core/src/lib/loader-utils/get-data.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ import {
   isBuffer
 } from '../../javascript-utils/is-type';
 import {makeIterator} from '../../iterator-utils/make-iterator/make-iterator';
-import {concatenateChunksAsync} from '../../iterator-utils/async-iteration';
+import {concatenateChunksAsync} from '@loaders.gl/loader-utils';
```
Comment (ibgreen) [inline modules/core/src/lib/loader-utils/get-data.js:12]: Nit: Usually put the imports from other modules at the top.

**Pull request merged**